pkg/sensors/tracing/kprobe_test.go

```diff
@@ -7717,3 +7717,70 @@ spec:
 	err = jsonchecker.JsonTestCheck(t, checker)
 	assert.NoError(t, err)
 }
+
+func TestKprobeArgsReverse(t *testing.T) {
+	var doneWG, readyWG sync.WaitGroup
+	defer doneWG.Wait()
+
+	ctx, cancel := context.WithTimeout(context.Background(), tus.Conf().CmdWaitTime)
+	defer cancel()
+
+	pidStr := strconv.Itoa(int(observertesthelper.GetMyPid()))
+	t.Logf("tester pid=%s\n", pidStr)
+
+	lseekConfigHook_ := `
+apiVersion: cilium.io/v1alpha1
+kind: TracingPolicy
+metadata:
+  name: "sys-write"
+spec:
+  kprobes:
+  - call: "sys_lseek"
+    return: false
+    syscall: true
+    args:
+    - index: 2
+      type: "int"
+      label: "index 2"
+    - index: 1
+      type: "int"
+      label: "index 1"
+    - index: 0
+      type: "int"
+      label: "index 0"
+    selectors:
+    - matchPIDs:
+      - operator: In
+        followForks: true
+        isNamespacePID: false
+        values:
+        - ` + pidStr
+
+	lseekConfigHook := []byte(lseekConfigHook_)
+	err := os.WriteFile(testConfigFile, lseekConfigHook, 0644)
+	if err != nil {
+		t.Fatalf("writeFile(%s): err %s", testConfigFile, err)
+	}
+
+	kpChecker := ec.NewProcessKprobeChecker("lseek-checker").
+		WithFunctionName(sm.Suffix("sys_lseek")).
+		WithArgs(ec.NewKprobeArgumentListMatcher().
+			WithOperator(lc.Ordered).
+			WithValues(
+				ec.NewKprobeArgumentChecker().WithIntArg(0).WithLabel(sm.Full("index 0")),
+				ec.NewKprobeArgumentChecker().WithIntArg(1).WithLabel(sm.Full("index 1")),
+				ec.NewKprobeArgumentChecker().WithIntArg(2).WithLabel(sm.Full("index 2")),
```

Comment:
In https://github.com/cilium/tetragon/pull/3730#discussion_r2086564108, you mentioned:
> with this change (still need to push final commit) the order should be given by the position in the spec

My interpretation of the position in the spec is that the arguments are = {"index 2", "index 1", "index 0"}

But this test (and, AFAICT the code)  indicates that the arguments are ordered based on the index value not the spec order. Having the arguments in "spec order" seems like a better choice for me. Also, I'm thinking that if we change the order, we might break users that depend on it.

Reply:
Let's continue the discussion in https://github.com/cilium/tetragon/issues/3711#issuecomment-2879269243

Reply:
so the order is always set by ebpf program from index 0,1... up to this change if you spcify different order, you will receive wrong data, because the printers will not match the ebpf program order

so this change fixes the printers order to match the ebpf argument retrieval and storage

Reply:
yap, I think I understand that part now. Thanks.

Since this fixes an obvious bug, we can merge it and then continue the discussion on what we want to do in https://github.com/cilium/tetragon/issues/3711#issuecomment-2879269243.